iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/AnonymousAccessToResourceCheck.java

```diff
@@ -82,7 +83,8 @@ private static void checkWebSitesAuthSettings(ContextualResource contextualResou
   }
 
   private static void checkApiManagementService(ContextualResource resource) {
-    ContextualResource signIn = resource.childResourceBy("portalsettings", it -> isValue(it.name(), APIMGMT_PORTALSETTINGS_SIGNIN_RESOURCE_NAME).isTrue());
+    ContextualResource signIn = resource.childResourceBy("portalsettings", it -> isValue(it.name(), APIMGMT_PORTALSETTINGS_SIGNIN_RESOURCE_NAME).isTrue() ||
+      isValue(it.name(), resource.name + "/" + APIMGMT_PORTALSETTINGS_SIGNIN_RESOURCE_NAME).isTrue());
 
```

Comment:
I'm not sure how to interpret this sentence from the ticket:
```
Also the only child resource of type Microsoft.ApiManagement/service/apis defined, contains an authenticationSettings property.
```
but maybe additional verification of `Microsoft.ApiManagement/service/apis` here is needed? 🤔

Reply:
We have a separate check for this resource (`AnonymousAccessToResourceCheck::checkApiManagementServiceApis`), and the issue should only be raised if it's misconfigured. We used to have this check inside this method, but then it was removed. I think in the current form it still represents the intended behavior.

Reply:
Actually, you are right; it seems I created a separate check for this resource to avoid duplication because at that time we didn't handle nested and separate children. Now it can be moved here again and handle both cases.

Reply:
Looks like I'll accidentally solve https://sonarsource.atlassian.net/browse/SONARIAC-1044 by this as well